### PR TITLE
171 create stripe routes and component

### PR DESCRIPTION
**Pull request by Cali0707**
This PR expanded a bit as I was working on it. It contains the code to connect the registration frontend to the backend, handle the payments, and allow users to login (but not sign up). You will need to update your .env file to test this, so please message me for details.

To test:
1. Create an account for yourself using the api
2. Login to that account
3. Go to the registration page
4. Use the test cards from (stripe)[https://stripe.com/docs/testing] to test if the payment works
5. Upon a successful payment you will be redirected back to the orientation site and it will say on the orientation site that an error occurred - that is the expected behaviour currently, as there is a separate ticket to fix that page.

### CODE REVIEW

File: server/src/services/PaymentServices.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        submit_type: 'pay',
+        billing_address_collection: 'auto',
+        line_items: [
+          {
+            price: process.env.STRIPE_TICKET_PRICE_ID,
+            quantity: 1,
+          },
+        ],
+        mode: 'payment',
+        success_url: 'http://localhost:3000/registration-success',
```
Comment (jameskokoska) [inline server/src/services/PaymentServices.js:61]: I assume this will be changed? from localshost

File: server/src/services/FroshServices.js [hunk 2]
```diff
@@ -31,11 +31,16 @@ const FroshServices = {
...
    * @return {Promise<Object>}
    */
-  async upgradeToFrosh(user, newInfo) {
+  async upgradeToFrosh(user, newInfo, paymentIntent) {
     const frosh = FroshModel.hydrate(user.toObject());
-    frosh.set({ ...newInfo, userType: 'frosh' });
+    frosh.set({
+      ...newInfo,
+      userType: 'frosh',
+      payments: [{ item: 'Orientation Ticket', paymentIntent, amountDue: 13000 }],
```
Comment (jameskokoska) [inline server/src/services/FroshServices.js:42]: does this handle the discount?

**Review by jameskokoska — APPROVED** [review]

File: server/src/services/FroshServices.js (hunk 2 above)
Comment (Cali0707) [reply server/src/services/FroshServices.js:42]: No, I'll update this later for discounts

File: server/src/services/PaymentServices.js (hunk 1 above)
Comment (Cali0707) [reply server/src/services/PaymentServices.js:61]: Yeah, good catch!

**Pull request merged**